packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1745,7 +1766,14 @@ actual object RealmInterop {
             staticCFunction { userData, beforeRealm ->
                 val beforeCallback = safeUserData<SyncBeforeClientResetHandler>(userData)
                 val beforeDb = CPointerWrapper<FrozenRealmT>(beforeRealm, false)
-                beforeCallback.onBeforeReset(beforeDb)
+
+                // Check if exceptions have been thrown, return true if all went as it should
+                try {
+                    beforeCallback.onBeforeReset(beforeDb)
+                    true
+                } catch (e: Throwable) {
+                    false
```

Comment:
Should we log the exception here? It seems to be swallowed otherwise?

Reply:
If this returns `false` then the execution follows through `onError` instead, which the intended behaviour we sought after reporting https://github.com/realm/realm-core/pull/5564 to Yavor.

Reply:
Yeah, we shouldn't throw it here, but this error is hidden so people wouldn't know they have buggy code. Could we at least do `println(e.toStrint)` here??

Reply:
Ah, sorry, I missed you meant log the error. Hm, we can do it easily for native but JVM is a bit more convoluted...

Reply:
I think we need to address JVM regardless, since exceptions are leaking. I think if we can use Kotlin println...it will work correctly on both JVM and Android...It isn't ideal, but should be sufficient until we can come up with something better

Reply:
Native is not an issue at all, I can call println with the exception message. For JVM I will add some helpers to extract the exception message from `api_helpers` from the JNI and then do the whole JNI thing to call `System.out.println`. I'm not sure we can call Kotlin's `println` since it's a top level function which might be tricky to find by JNI.